packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -169,9 +174,44 @@ - (instancetype)initWithCameraName:(NSString *)cameraName
   _motionManager = [[CMMotionManager alloc] init];
   [_motionManager startAccelerometerUpdates];
 
-  if (![self setCaptureSessionPreset:_resolutionPreset withError:error]) {
-    return nil;
+  NSError *outError = nil;
+
+  if (_mediaSettingsProvider.fps) {
+    // The frame rate can be changed only on a locked for configuration device.
+    if ([mediaSettingsProvider lockDevice:_captureDevice error:&outError]) {
+      [_mediaSettingsProvider beginConfigurationForSession:_videoCaptureSession];
+
+      // Possible values for presets are hard-coded in FLT interface having
+      // corresponding AVCaptureSessionPreset counterparts.
+      // If _resolutionPreset is not supported by camera there is
+      // fallback to lower resolution presets.
+      // If none can be selected there is error condition.
+      if (![self setCaptureSessionPreset:_resolutionPreset withError:error]) {
+        [_videoCaptureSession commitConfiguration];
+        [_captureDevice unlockForConfiguration];
+        return nil;
+      }
+
+      int fpsNominator = floor([_mediaSettingsProvider.fps doubleValue] * 10.0);
+      CMTime duration = CMTimeMake(10, fpsNominator);
+
+      [mediaSettingsProvider setMinFrameDuration:duration onDevice:_captureDevice];
+      [mediaSettingsProvider setMaxFrameDuration:duration onDevice:_captureDevice];
+
+      [_mediaSettingsProvider commitConfigurationForSession:_videoCaptureSession];
+      [_mediaSettingsProvider unlockDevice:_captureDevice];
+    } else if (outError) {
+      *error = outError;
```

Comment:
You added a check that `error` isn't NULL before using it in the existing code above, but are using it unconditionally here.

(This issue goes away if you use `error`.)

Reply:
`outError` eliminated following previous suggestion.